### PR TITLE
Generate license headers through the JVM only

### LINKED ISSUE #220: Sometimes genEnums generates bogus code
Sometimes when running `genEnums` extra bogus code snippets get inserted. See the compile error below as an example but the generated snippets are not always the same.

Running `genEnums` multiple times fixes it.

It sounds like an upstream error (sbt?) but I'm reporting it here to keep track of it.

```
[error] /Users/jnavarro/Develop/gem/modules/core/shared/src/main/scala/gem/enum/GcalLampType.scala:45:12: expected start of definition
[error]   implicit val GcalLampTypeEnumerated: Enumerated[GcalLampType] =
[error]            ^
[error] /Users/jnavarro/Develop/gem/modules/core/shared/src/main/scala/gem/enum/GcalLampType.scala:45:12: expected start of definition
[error]   implicit val GcalLampTypeEnumerated: Enumerated[GcalLampType] =
[error]            ^
[error] /Users/jnavarro/Develop/gem/modules/core/shared/src/main/scala/gem/enum/GcalLampType.scala:51:1: Unmatched closing brace '}' ignored here
[error] }
[error] ^
[error] /Users/jnavarro/Develop/gem/modules/core/shared/src/main/scala/gem/enum/GcalLampType.scala:51:1: Unmatched closing brace '}' ignored here
[error] }
[error] ^
[error] two errors found
[error] two errors found
```

### PR DESCRIPTION
**Pull request by jdnavarro**
Fixes #220.

### CODE REVIEW

**Comment by cquiroz** [comment]
could it be better to fix it by removing the plugin from `coreJS`. That should work on all cases rather than only on the custom command

**Comment by swalker2m** [comment]
> could it be better to fix it by removing the plugin from coreJS. That should work on all cases rather than only on the custom command

Agree, that would be ideal.

**Comment by swalker2m** [comment]
> Oddly I have never had a problem with this but if it fixes your issue 

Don't you usually skip the scala.js build?  If so, that may be why.

**Comment by tpolecat** [comment]
Yeah I spend most of my time in `coreJVM` and `db` but I didn't think that would affect a command since they're defined globally. I don't understand sbt.

**Comment by jdnavarro** [comment]
I still have this issue with the latest commit I pushed. After trying several ways of disabling `automateHeaderPlugin` for JS, I've only seen this problem go away consistently in my original first commit, when I place `coreJVM/automateHead` in the `genEnums` alias.

I can keep trying different combinations of disabling the plugin if you still don't like that solution.

**Comment by swalker2m** [comment]
In addition to the current version of this PR, you might also remove the header generation after the enum generation. Headers will be added by the compile step.  Could you give that a try?

So, also update the line:

```
addCommandAlias("genEnums", "; sql/runMain gem.sql.Main modules/core/shared/src/main/scala/gem/enum; headerCreate")
```

to

```
addCommandAlias("genEnums", "; sql/runMain gem.sql.Main modules/core/shared/src/main/scala/gem/enum")
```

**Comment by swalker2m** [comment]
In my experience, the problem occurs when the set of enums are divided into two groups at the time headers are generated.  One or more of the enums appears in both groups, and those are the ones that end up with corrupt source files.  Is that the case for you when you experience the problem too?  If so, I assume the command is somehow still being run simultaneously by the normal build and the JS build.

The biggest issue will be with the enums because it is a large set of source files that need the headers all at once.  If the problem does happen with a normal source file though it will be annoying because you won't be able to just regenerate it.

**Comment by cquiroz** [comment]
This could be tested by removing the headers and run `coreJS/compile` and verify the headers are created

**Comment by swalker2m** [comment]
> This could be tested by removing the headers and run `coreJS/compile` and verify the headers are created

In the current version of this PR, my theory is that it is happening as a result of the `genEnums` alias.

**Comment by cquiroz** [comment]
If adding headers is implicit with compile we shouldn't include it in the alias. I think

**Comment by jdnavarro** [comment]
>  One or more of the enums appears in both groups, and those are the ones that end up with corrupt source files. Is that the case for you when you experience the problem too?

I've seen all kinds of corrupt enumerations, but I think this is the case for me too.

**Comment by jdnavarro** [comment]
If I remove the header generation command from the alias, no headers get generated in the enumerations. I also tried different ways of enabling the plugin below too, but no luck.

**Comment by cquiroz** [comment]
i have never seen this happening in ocs3 with crossprojects. I suppose because we don't generate code

**Comment by jdnavarro** [comment]
> This could be tested by removing the headers and run coreJS/compile and verify the headers are created.

`coreJS/headersCreate` also fixes it but I keep seeing the 2 groups of enumerations being generated.

What it's fixing it with just 1 group is just disabling the plugin for the core project, and using `headersCreate` in the `genEmum` alias, as it was before.

File: build.sbt [hunk 1]
```diff
@@ -36,7 +36,7 @@ onLoad in Global := { s =>
 cancelable in Global := true
 
 // some extra commands for us
-addCommandAlias("genEnums", "; sql/runMain gem.sql.Main modules/core/shared/src/main/scala/gem/enum; headerCreate")
```
Comment (swalker2m) [inline build.sbt]: I think we can leave the compile off and just let that be a separate step.  The headers will be added then just as for everything else.

Comment (tpolecat) [reply build.sbt]: That's fine with me.

**Pull request merged**